jwst/cube_build/ifu_cube.py

```diff
@@ -487,6 +494,7 @@ def build_ifucube(self):
         self.spaxel_flux = np.zeros(total_num)
         self.spaxel_weight = np.zeros(total_num)
         self.spaxel_iflux = np.zeros(total_num)
+        self.spaxel_dq = np.zeros((self.naxis3, self.naxis2 * self.naxis1), dtype=np.int32)
```

Comment:
Should the dtype be **u**int32 (unsigned int32)? If it's signed, you lose the use of 1 bit and hence have a max of only 31 values.

Reply:
made change